inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp

```diff
@@ -0,0 +1,330 @@
+    return std::make_shared<ngraph::opset7::FakeQuantize>(parent_node, input_low,
+                                                          input_high, output_low,
+                                                          output_high, 11);
+}
+
+class CreateConcat : public CreateGraphDecorator {
+public:
+    CreateConcat(CreateGraphDecoratorPtr prev) :
+        CreateGraphDecorator(std::move(prev)) {}
+protected:
+    void updateGraph(Graph& graph) override;
+};
+
+void CreateConcat::updateGraph(Graph& graph) {
+    ngraph::OutputVector new_graph_output;
+    new_graph_output.emplace_back(std::make_shared<ngraph::opset7::Concat>(graph.output_nodes, 3));
+    graph.output_nodes.swap(new_graph_output);
+}
+
+// -------------------------------------------------------------------------------------------------------
+
+template<typename DecorT, typename... DecorTs>
+auto createBuildDecorator() -> typename std::enable_if<(sizeof...(DecorTs) == 0), CreateGraphDecoratorPtr>::type {
+    CreateGraphDecoratorPtr build_decorator = createUnique<CreateBaseDecorator>();
+    return createUnique<DecorT>(std::move(build_decorator));
+}
+
+template<typename DecorT, typename... DecorTs>
+auto createBuildDecorator() -> typename std::enable_if<(sizeof...(DecorTs) > 0), CreateGraphDecoratorPtr>::type {
+    CreateGraphDecoratorPtr build_decorator = createBuildDecorator<DecorTs...>();
+    return createUnique<DecorT>(std::move(build_decorator));
+}
+
+template<typename DecorT, typename... DecorTs>
+Graph createGraph() {
+    CreateGraphDecoratorPtr build_decorator = createBuildDecorator<DecorT, DecorTs...>();
+    return build_decorator->build();
+}
+
+CreateGraphDecoratorPtr createBuildDecorator(const ngraph::Shape& input_shape, const ngraph::Shape& kernel_shape) {
+    CreateGraphDecoratorPtr base_decorator = createUnique<CreateBaseDecorator>(input_shape);
+    return createUnique<CreateConvolution>(std::move(base_decorator), kernel_shape);
+}
+
+template<typename DecorT, typename... DecorTs>
+auto createBuildDecorator(const ngraph::Shape& input_shape,
+                                             const ngraph::Shape& kernel_shape)
+                                                -> typename std::enable_if<(sizeof...(DecorTs) == 0), CreateGraphDecoratorPtr>::type {
+    CreateGraphDecoratorPtr build_decorator = createBuildDecorator(input_shape, kernel_shape);
+    return createUnique<DecorT>(std::move(build_decorator));
+}
+
+template<typename DecorT, typename... DecorTs>
+auto createBuildDecorator(const ngraph::Shape& input_shape,
+                                             const ngraph::Shape& kernel_shape)
+                                                -> typename std::enable_if<(sizeof...(DecorTs) > 0), CreateGraphDecoratorPtr>::type {
+    CreateGraphDecoratorPtr build_decorator = createBuildDecorator<DecorTs...>(input_shape, kernel_shape);
+    return createUnique<DecorT>(std::move(build_decorator));
+}
```

Comment:
Consider to split template conditions and return types: https://godbolt.org/z/conbefe6s